### ISSUE TITLE
legacy_constructor autocorrect doesn't correct all cases

### ISSUE THREAD

**Issue by fbarraganef** [realm/SwiftLint#2098]
### New Issue Checklist

- [ x] Updated SwiftLint to the latest version
- [ x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report
When I execute the swiftlint command, It show me the detected errors and when I execute the swiftlint autocorrect it show that all found errors were corrected, but when I open the file again the issue is persisting.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Done linting! Found 10 violations, 0 serious in 146 files.

swiftlint autocorrect 
Done correcting 146 files!
```
### Environment
* SwiftLint version (run `swiftlint version` to be sure)?
0.25.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
homebrew

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
NO
* Which Xcode version are you using (check `xcode-select -p`)? 9.2

```swift
// This triggers a violation:
static func strikePrice(value: String) -> NSAttributedString { let attributeString: NSMutableAttributedString = NSMutableAttributedString(string: value) attributeString.addAttribute(NSBaselineOffsetAttributeName, value: 0, range: NSMakeRange(0, attributeString.length)) attributeString.addAttribute(NSStrikethroughStyleAttributeName, value: 1, range: NSMakeRange(0, attributeString.length)) return attributeString }
```
